File: Remove Duplicates from Sorted Array/remove_duplicates_from_sorted_array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 var removeDuplicates = function(nums) {
  for(n=1; n<nums.length; n++) {
      console.log(nums)
      if(nums[n] === nums[n-1]) {
        nums.splice(n,1)
        n = n - 1
      }       
  }
};

console.log(removeDuplicates([0,0,1,1,1,2,2,3,3,3,4]))
console.log(removeDuplicates([1,1,2]))

File: Remove Duplicates from Sorted Array/remove_duplicates_from_sorted_array.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        nonDuplicateIndex = 0
        for i in range(len(nums)):
            if i+1 < len(nums) and nums[i] == nums[i+1]:
                continue
            else:
                nums[nonDuplicateIndex] = nums[i]
                nonDuplicateIndex += 1
        return nonDuplicateIndex


classInstance = Solution()
print(classInstance.removeDuplicates([0,0,1,1,1,2,2,3,3,3,4]))
print(classInstance.removeDuplicates([1,1,2]))